metron-platform/metron-indexing/src/test/java/org/apache/metron/indexing/dao/InMemoryDao.java

```diff
@@ -221,12 +222,24 @@ public void batchUpdate(Map<Document, Optional<String>> updates) throws IOExcept
     }
   }
 
-  public Map<String, Map<String, FieldType>> getColumnMetadata(List<String> indices) throws IOException {
-    Map<String, Map<String, FieldType>> columnMetadata = new HashMap<>();
+  @Override
+  public Map<String, FieldType> getColumnMetadata(List<String> indices) throws IOException {
+    Map<String, FieldType> indexColumnMetadata = new HashMap<>();
     for(String index: indices) {
-      columnMetadata.put(index, new HashMap<>(COLUMN_METADATA.get(index)));
+      Map<String, FieldType> columnMetadata = COLUMN_METADATA.get(index);
+      for (Entry entry: columnMetadata.entrySet()) {
+        String field = (String) entry.getKey();
+        FieldType type = (FieldType) entry.getValue();
+        if (indexColumnMetadata.containsKey(field)) {
+          if (!type.equals(indexColumnMetadata.get(field))) {
+            indexColumnMetadata.remove(field);
```

Comment:
>  The case of columns existing in multiple indices with different types is solved by just excluding them from the results.

Why did you choose to exclude these fields?  

What happens If the user adds a new sensor that triggers this condition?  They add a sensor that has the same name, but different type.  Wouldn't the user no longer be able to see that field in the UI?   That seems like it would be confusing for the user to understand what has happened.

Reply:
I agree with you that it would be confusing but I'm not sure what the correct behavior should be.  Should we include the field but just set the type to OTHER?  This is how the ElasticsearchDao treats fields it doesn't have type information for but so it might be better to explicitly state this in the column metadata endpoint response.

Reply:
> I agree with you that it would be confusing but I'm not sure what the correct behavior should be. 

Right.  I don't think there is an obviously correct answer here. 

Whichever approach we do take, I think we should have a big fat, `LOG.error` here logging the duplicate field names, the index names, and what the impact of this is.

> Should we include the field but just set the type to OTHER? This is how the ElasticsearchDao treats fields it doesn't have type information for but so it might be better to explicitly state this in the column metadata endpoint response.

Do you know how the UI behaves if we were to do this?  I think your idea definitely sounds reasonable, but I'd want to see how the UI behaves with the logic defined like this.

Reply:
And actually my comments here are tagged on the `InMemoryDao`.  I should have started this conversation on the `ElasticsearchDao`, not here.

Reply:
As far as I know the only place the UI uses this endpoint directly is when it presents a list of fields a user can select to display in the UI.  I think we would want it to show up there so including it with a type set to OTHER makes sense to me. 

If you're ok with it I will proceed with this change and add a big fat `LOG.error` call.

Reply:
> As far as I know the only place the UI uses this endpoint directly is when it presents a list of fields a user can select to display in the UI.

It is also used for sorting and faceting (but that is internal to the ElasticsearchDAO.)

Reply:
> If you're ok with it I will proceed with this change and add a big fat LOG.error call.

Yes, let's give it a go.